src/main/kotlin/pl/edu/agh/gem/internal/service/UserDetailsService.kt

```diff
@@ -22,6 +23,23 @@ class UserDetailsService(
     fun getUserDetails(userId: String): UserDetails {
         return userDetailsRepository.findById(userId) ?: throw MissingUserDetailsException(userId)
     }
+
+    fun updateUserDetails(userDetailsUpdate: UserDetailsUpdate) {
+        val userDetails = userDetailsRepository.findById(userDetailsUpdate.userId) ?: throw MissingUserDetailsException(userDetailsUpdate.userId)
+
+        val updatedUserDetails = UserDetails(
+            id = userDetails.id,
+            username = userDetailsUpdate.username ?: userDetails.username,
+            firstName = userDetailsUpdate.firstName ?: userDetails.firstName,
+            lastName = userDetailsUpdate.lastName ?: userDetails.lastName,
+            phoneNumber = userDetailsUpdate.phoneNumber ?: userDetails.phoneNumber,
```

Comment:
what update case when we want to remove phoneNumber or sth else?

Reply:
maybe when use doesnt't want to modify given field app will send null for this field. and if user wants to remove fielfd's value they pass empty string. What do you tihnk?

Reply:
second option is to always update - when new value is different ffrom old we put new value. when new value is the same as old one we also update, when new value is null we put null

Reply:
i prefer second option, what do u think?

Reply:
sure we can do that in the second scenario

Reply:
done